File: Lab 6/t4.go
package main

import (
  "fmt"
  "os"
  "bufio"
)

func main() {
  os.Exit(t4_main())
}

func t4_main() int {
  var nline int
  var e error

  stdin := bufio.NewReader(os.Stdin)

  fmt.Print("How many line shall we draw? ")
  _, e = fmt.Fscanf(stdin, "%d", &nline)
  if e != nil {
    fmt.Println("Illegal input:", e)
    return 1
  }

  for row := 0; row < nline; row += 1 {
    to_print_asterisk := row * 2 + 1
    to_print_blank := (((nline - 1) * 2 + 1) - to_print_asterisk) / 2 + 1

    for i := 0; i < to_print_blank; i += 1 {
      fmt.Print(" ")
    }
    for i := 0; i < to_print_asterisk; i += 1 {
      fmt.Print("*")
    }

    fmt.Print("\n")
  }

  return 0
}

File: Lab 6/t4.rs
use std::io::Write;

fn main() {
    print!("How many line shall we draw? ");
    std::io::stdout().flush().unwrap();
    let mut buf = String::new();
    match std::io::stdin().read_line(&mut buf) {
        Ok(_) => {},
        Err(e) => {
            println!("Read error: {}", e);
            std::process::exit(1);
        }
    };

    let nline = match buf.split_whitespace().next() {
        Some(r) => {
            match r.parse::<i32>() {
                Ok(r) => r,
                Err(e) => {
                    println!("Parse error: {}", e);
                    std::process::exit(1);
                }
            }
        },
        None => {
            println!("No input is given... Exiting.");
            return;
        }
    };

    for row in 0..nline {
        let to_print_asterisk = row * 2 + 1;
        let to_print_blank = (((nline - 1) * 2 + 1) - to_print_asterisk) / 2 + 1;

        for _ in 0..to_print_blank {
            print!(" ");
        }
        for _ in 0..to_print_asterisk {
            print!("*");
        }
        println!("");
    }
}
